bracket-pathfinding/src/dijkstra.rs

```diff
@@ -99,65 +95,44 @@ impl DijkstraMap {
     }
 
     #[cfg(feature = "threaded")]
-    fn build_helper(dm: &mut DijkstraMap, starts: &[usize], map: &dyn BaseMap) {
-        if starts.len() > rayon::current_num_threads() {
+    fn build_helper(dm: &mut DijkstraMap, starts: &[usize], map: &dyn BaseMap) -> RunThreaded {
+        if starts.len() >= THREADED_REQUIRED_STARTS {
             DijkstraMap::build_parallel(dm, starts, map);
-            return;
+            return RunThreaded::True;
         }
+        RunThreaded::False
     }
 
     #[cfg(not(feature = "threaded"))]
-    fn build_helper(_dm: &mut DijkstraMap, _starts: &[usize], _map: &dyn BaseMap) {}
+    fn build_helper(_dm: &mut DijkstraMap, _starts: &[usize], _map: &dyn BaseMap) -> RunThreaded {
+        RunThreaded::False
+    }
 
     /// Builds the Dijkstra map: iterate from each starting point, to each exit provided by BaseMap's
     /// exits implementation. Each step adds cost to the current depth, and is discarded if the new
     /// depth is further than the current depth.
-    /// If you provide more starting points than you have CPUs, automatically branches to a parallel
-    /// version.
+    /// WARNING: Will give incorrect results when used with non-uniform exit costs. Much slower
+    /// algorithm required to support that.
+    /// Automatically branches to a parallel version if you provide more than 4 starting points
     pub fn build(dm: &mut DijkstraMap, starts: &[usize], map: &dyn BaseMap) {
-        DijkstraMap::build_helper(dm, starts, map);
+        let threaded = DijkstraMap::build_helper(dm, starts, map);
+        if threaded == RunThreaded::True { return; }
         let mapsize: usize = (dm.size_x * dm.size_y) as usize;
-        let mut open_list: Vec<(usize, f32)> = Vec::with_capacity(mapsize * 2);
-        let mut closed_list: Vec<bool> = vec![false; mapsize];
+        let mut open_list: VecDeque<(usize, f32)> = VecDeque::with_capacity(mapsize);
```

Comment:
This initial capacity can probably be lowered a lot, since the open list will be a wave-front travelling over the map in a BFS manner. Now that I think of it... For a square tiled map the most degenerate case would use 8/9 of mapsize. For most normal cases it'd probably be somewhere on the order of `max(width, height) + min(width, height)/2`. Not sure what the perf impact would be of changing it, and I'm too tired to test at the moment.